bw2data/project.py

```diff
@@ -35,6 +39,10 @@ def lockable():
 
 
 class ProjectDataset(Model):
+    # Event sourcing
+    is_sourced = BooleanField(default=False, constraints=[SQL("DEFAULT 0")])
+    revision = TextField(null=True, default=uuid.uuid4)
```

Comment:
Maybe instead:
```python
def uuid4_func() -> str:
    return uuid.uuid4().hex
    
revision = TextField(null=True, default=uuid4_func)
```

I guess this would work but the implicit type conversion feels icky.

Reply:
stale, we use snowflake id